### PR TITLE
Align atom on arbitrary axis and addition of new centering methods

### LINKED ISSUE #166: Issue with atom.center
**Describe the bug**
There is an indexing issue in the `exatomic.atom.Atom.center` method. I believe that this comes from the way that we are currently trying to grab the index with `pandas.DataFrame.ix` as it is an index dependent lookup. It works as expected when we set `frame=0`, but anything else raises a `KeyError` exception.
I propose that we change `center = frame.ix[idx]` to `center = frame.iloc[idx]`. This is mentioned in the pandas docs

> A scalar index that is not found will raise a KeyError. Slicing is primarily on the values of the index when using [],ix,loc, and always positional when using iloc. The exception is when the slice is boolean, in which case it will always be positional.

**To Reproduce**
Minimal example to get the bug
```
from exatomic.base import resource
from exatomic.gaussian import Output

ed = Output(resource('g16-methyloxirane-def2tzvp-opt.out'))
ed.parse_atom()
ed.atom.center(idx=0)
```

**Expected behavior**
Returns the last frame with the first atom in the molecule centered on the origin.

**Desktop (please complete the following information):**
 - OS: WSL Ubuntu

### LINKED ISSUE #167: Align molecule along an axis and new centering methods
**Is your feature request related to a problem? Please describe.**
We have already implemented methods to be able to center and rotate a molecule. I think it would not be a big step to add a method to center and align a molecule along some pre-defined axis.

**Describe the solution you'd like**
This should be fairly simple and the pseudo code would probably look like this:
1). Define the vector going through the atoms that are to be aligned
2). Get the normal vector from the two vectors with the vector product
3). Get the angle with the scalar product
4). Center the molecule with the `Atom.center` method
5). Rotate the molecule with the `Atom.rotate` method

### PR DESCRIPTION
**Pull request by herbertludowieg**
This closes #166 and closes #167.

Fixed bug in `Atom.center`

Additions:
- New code to center the molecule to the center of nuclear charge
- Align molecule along an arbitrary axis
   - This will also center the molecule
- Some possible zero division error handling when normalizing vector in `Atom.rotate`

### CODE REVIEW

**Comment by herbertludowieg** [comment]
Should we consider adding `return Atom(return_var)` for the `center`, `rotate`, `translate` and `align` methods. As it is right now it only returns an instance of `pandas.DataFrame` instead of `Atom`.

**Comment by herbertludowieg** [comment]
I just noticed that the new `Atom.align` method may have overwritten a pre-existing method in `pandas.DataFrame`. Also, I would like to add tests to this PR.

**Comment by avmarchenko** [comment]
Given your last comment, lets rename the method so that it doesn't mess with pandas API.

**Comment by herbertludowieg** [comment]
I agree I'll rename it to align_mol. Seems to make enough sense to me.

**Comment by avmarchenko** [comment]
@herbertludowieg , lets merge master into this PR, resolve conflicts, and then see what the build system reports.

File: exatomic/core/atom.py [hunk 1]
```diff
@@ -105,6 +139,8 @@ def rotate(self, theta, axis=None, frame=None, degrees=True):
         if axis is None: axis = [0, 0, 1]
         if frame is None: frame = self.last_frame.copy()
         else: frame = self[self.frame == frame].copy()
+
+        if all(list(map(lambda x: x == 0., axis))) or theta == 0.: return frame
```
Comment (avmarchenko) [inline exatomic/core/atom.py]: Does `all(map(lambda x: x...))`?

Comment (herbertludowieg) [reply exatomic/core/atom.py]: Yeah, that would probably look better. I pushed the updates. Thank you!

**Pull request merged**